codalab/lib/upload_manager.py

```diff
@@ -503,3 +515,76 @@ def upload_Azure_blob_storage(
             index_conn_str,
             progress_callback,
         )
+
+    def upload_GCS_blob_storage(
+        self,
+        fileobj,
+        bundle_url,
+        bundle_conn_str,
+        bundle_read_str,
+        index_conn_str,
+        source_ext,
+        should_unpack,
+        progress_callback=None,
+    ):
+        from codalab.lib import zip_util
+
+        if should_unpack:
+            output_fileobj = zip_util.unpack_to_archive(source_ext, fileobj)
+        else:
+            output_fileobj = GzipStream(fileobj)
+
+        # Write archive file.
+        self._upload_with_chunked_encoding(
+            method='PUT',
+            url=bundle_conn_str,
+            header={'Content-type': 'application/octet-stream'},
+            fileobj=output_fileobj,
+            progress_callback=progress_callback,
+        )
+        # upload the index file
+        with httpopen_with_retry(bundle_read_str) as ttf, tempfile.NamedTemporaryFile(
+            suffix=".sqlite"
+        ) as tmp_index_file:
+            SQLiteIndexedTar(
+                fileObject=ttf,  # TODO: This need to load the file into memory
+                tarFileName="contents",  # If saving a single file as a .gz archive, this file can be accessed by the "/contents" entry in the index.
+                writeIndex=True,
+                clearIndexCache=True,
+                indexFilePath=tmp_index_file.name,
+            )
+            self._upload_with_chunked_encoding(
+                method='PUT',
+                url=index_conn_str,
+                header={'Content-type': 'application/octet-stream'},
+                fileobj=open(tmp_index_file.name, "rb"),
+                progress_callback=None,
+            )
+
+    def _upload_with_chunked_encoding(self, method, url, header, fileobj, progress_callback=None):
```

Comment:
Is this the same function that is found in rest_client.py? If so, should we find a way to unify them?